pkg/storage/stores/shipper/bloomshipper/shipper.go

```diff
@@ -4,32 +4,35 @@ import (
 	"cmp"
 	"context"
 	"fmt"
-	"io"
-	"os"
-	"path/filepath"
-	"strconv"
-	"strings"
 	"time"
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/exp/slices"
 
-	v1 "github.com/grafana/loki/pkg/storage/bloom/v1"
 	"github.com/grafana/loki/pkg/storage/stores/shipper/bloomshipper/config"
 )
 
 type Shipper struct {
-	client Client
-	config config.Config
-	logger log.Logger
+	client          Client
+	config          config.Config
+	logger          log.Logger
+	blockDownloader *blockDownloader
 }
 
-func NewShipper(client Client, config config.Config, logger log.Logger) (*Shipper, error) {
+type Limits interface {
+	BloomGatewayBlocksDownloadingParallelism(tenantID string) int
```

Comment:
Could be more concise IMO

```suggestion
	BloomShipperDownloadParallelism(tenantID string) int
```

Reply:
yes, it is also correct, but `BloomGatewayBlocksDownloadingParallelism` is more descriptive and explains immediately that it's an option that affects exactly blocks downloading parallelism.  so, I would leave it as is if it's not critical ;)